ci_feature/main.sh

```diff
@@ -288,7 +307,13 @@ cat /var/opt/microsoft/docker-cimprov/state/containerhostname
 rm -f /etc/opt/microsoft/omsagent/conf/omsagent.d/omsconfig.consistencyinvoker.conf
 
 if [ -z $INT ]; then
-  if [ -a /etc/omsagent-secret/DOMAIN ]; then
+  if [ -a /etc/omsagent-secret/PROXY ]; then
+     if [ -a /etc/omsagent-secret/DOMAIN ]; then
+        /opt/microsoft/omsagent/bin/omsadmin.sh -w `cat /etc/omsagent-secret/WSID` -s `cat /etc/omsagent-secret/KEY` -d `cat /etc/omsagent-secret/DOMAIN` -p `cat /etc/omsagent-secret/PROXY`
```

Comment:
why does this need to be secret? This is getting written to omsadmin.conf anyways, now its as secret as well ? why ? Also who will put this as secret ? chart ?

Reply:
> why does this need to be secret? This is getting written to omsadmin.conf anyways, now its written as secret as well ? why ? Also who will put this as secret ? chart ?

> since the proxy can have creds in it. the pattern is same as what we do for WSID, KEY and DOMAIN.  Parameter from chart will be given to the secret.  omsadmin.conf will not have the Proxy rather base agent puts into proxy.conf.

Reply:
if its in proxy.conf, why does it needs to be a secret ? If i change the secret object, there is no one watching & restarting the pod. I see no reason why it has to be in 2 places. Just set it as env (& proxy.conf by omsagent) in linux, and env as windows and use it. Storing in .conf anyways defeats the purpose of secret. Its just redundant and one more thing to manage.

Reply:
> if its in proxy.conf, why does it needs to be a secret ? If i change the secret object, there is no one watching & restarting the pod. I see no reason why it has to be in 2 places. Just set it as env (& proxy.conf by omsagent) in linux, and env as windows and use it. Storing in .conf anyways defeats the purpose of secret. Its just redundant and one more thing to manage.
 
The same pattern as we do for WSID, KEY and Domain. Here is the flow , proxy will be coming from secret similar to WSID., KEY and DOMAIN which in turn will be populated by the chart. Base agent omsadmin.sh, puts the WSID and DOMAIN etc .. to the omsadmin.conf but where as proxy puts into the proxy.conf.  If we want to avoid putting mutiple places (secret and conf(admin or proxy)) then we need to make changes in baseagent not put into conf files at all so that all these can be read from secrets itself.